qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -128,11 +134,15 @@ def __repr__(self):
 
     def __eq__(self, other):
         """Check if two SparsePauliOp operators are equal"""
-        return (
-            super().__eq__(other)
-            and np.allclose(self.coeffs, other.coeffs)
-            and self.paulis == other.paulis
-        )
+        close_coeffs = []
+        for i in range(self.coeffs.shape[0]):
+            # Check for Parameters separately
+            if isinstance(self.coeffs[i], ParameterExpression):
+                close_coeffs.append(self._coeffs[i] == other._coeffs[i])
+            else:
+                close_coeffs.append(np.isclose(self.coeffs[i], other.coeffs[i]))
```

Comment:
There's no need to build up a list here - as soon as you find _any_ element that's `False`, and you don't need to remember any `True` results.